src/calculus/HadamardProd.jl

```diff
@@ -0,0 +1,109 @@
+#HadamardProd
+
+export HadamardProd
+
+"""
+`HadamardProd(A::AbstractOperator,B::AbstractOperator)`
+
+Compose opeators such that:
+
+`(Ax).*(Bx)`
```

Comment:
This passage is not really clear. Maybe mention explicitly state that P*x = A*x .* B*x, where P is the operator being constructed.

Also, this operator seems to in fact reduce to the usual Hadamard product between matrices in the linear case: is the nonlinear case a well known generalization? Just asking out of curiosity, to make sure there’s no confusion with the naming

Reply:
I'll make the docs more clear, thanks for pointing that out!

Not sure how much the nonlinear generalization is known, however it is as usual the chain rule applied 😄 

Anyway the name could be perhaps improved with something like `HadamardCompose`, but it is very lengthy. 

By the way my idea is to deprecate the other calc rule `Hadamard` since `HadamardProd` generalises it. So definitively it won't be confused with that one.